llama_stack/providers/inline/post_training/torchtune/utils.py

```
@@ -0,0 +1,110 @@
+        "checkpoint_type": "LLAMA3_2",
+    },
+    "Llama-3-8B-Instruct": {
+        "model_definition": lora_llama3_8b,
+        "tokenizer_type": llama3_tokenizer,
+        "checkpoint_type": "LLAMA3",
+    },
+}
+
+EXPECTED_DATASET_SCHEMA: Dict[str, List[Dict[str, ParamType]]] = {
+    "alpaca": [
+        {
+            ColumnName.instruction.value: StringType(),
+            ColumnName.input.value: StringType(),
+            ColumnName.output.value: StringType(),
+            ColumnName.text.value: StringType(),
+        },
+        {
+            ColumnName.instruction.value: StringType(),
+            ColumnName.input.value: StringType(),
+            ColumnName.output.value: StringType(),
+        },
+        {
+            ColumnName.instruction.value: StringType(),
+            ColumnName.output.value: StringType(),
+        },
+    ]
+}
+
+BuildLoraModelCallable = Callable[..., torch.nn.Module]
+BuildTokenizerCallable = Callable[..., Llama3Tokenizer]
+
+
+async def get_model_definition(
+    model_id: str,
+) -> BuildLoraModelCallable:
+    model = resolve_model(model_id)
+    if model is None or model.core_model_id.value not in MODEL_CONFIGS:
+        raise ValueError(f"Model {model_id} is not supported.")
+    return MODEL_CONFIGS[model.core_model_id.value]["model_definition"]
+
+
+async def get_tokenizer_type(
+    model_id: str,
+) -> BuildTokenizerCallable:
+    model = resolve_model(model_id)
+    return MODEL_CONFIGS[model.core_model_id.value]["tokenizer_type"]
+
+
+async def get_checkpointer_model_type(
+    model_id: str,
+) -> str:
+    """
+    checkpointer model type is used in checkpointer for some special treatment on some specific model types
+    For example, llama3.2 model tied weights (https://github.com/pytorch/torchtune/blob/main/torchtune/training/checkpointing/_checkpointer.py#L1041)
+    """
+    model = resolve_model(model_id)
+    return MODEL_CONFIGS[model.core_model_id.value]["checkpoint_type"]
```

can you raise ValueError if the model key is not found? same thing in `get_tokenizer_type`